### PR TITLE
imgbrd-grabber: 7.10.0 -> 7.12.2

### LINKED ISSUE #312422: Update request: imgbrd-grabber 7.10.0 → 7.12.2
- Package name: imgbrd-grabber
- Latest released version: [7.12.2](https://github.com/Bionus/imgbrd-grabber/releases/tag/v7.12.2)

- Current version on the unstable channel: 7.10.0
- Current version on the stable/release channel: 7.10.0

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@evanjs

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #312422

Also removed entry from `top-level` package folder and moved it to `pkgs/by-name` folder accordingly to the newer schema

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by luftmensch-luftmensch** [comment]
```
> nix build .#imgbrd-grabber
> tree result/
 result
├──  bin
│  ├──  .Grabber-cli-wrapped -> ../share/Grabber/Grabber-cli
│  ├──  .Grabber-wrapped -> ../share/Grabber/Grabber
│  ├──  Grabber
│  └──  Grabber-cli
├──  include
│  └──  lexbor
│     ├──  core
│     │  ├──  array.h
│     │  ├──  array_obj.h
│     │  ├──  avl.h
│     │  ├──  base.h
│     │  ├──  bst.h
│     │  ├──  bst_map.h
│     │  ├──  conv.h
│     │  ├──  core.h
│     │  ├──  def.h
│     │  ├──  diyfp.h
│     │  ├──  dobject.h
│     │  ├──  dtoa.h
│     │  ├──  fs.h
│     │  ├──  hash.h
│     │  ├──  in.h
│     │  ├──  lexbor.h
│     │  ├──  mem.h
│     │  ├──  mraw.h
│     │  ├──  perf.h
│     │  ├──  plog.h
│     │  ├──  print.h
│     │  ├──  sbst.h
│     │  ├──  serialize.h
│     │  ├──  shs.h
│     │  ├──  str.h
│     │  ├──  str_res.h
│     │  ├──  strtod.h
│     │  ├──  types.h
│     │  └──  utils.h
│     ├──  css
│     │  ├──  selectors
│     │  │  ├──  base.h
│     │  │  ├──  pseudo.h
│     │  │  ├──  pseudo_const.h
│     │  │  ├──  pseudo_res.h
│     │  │  ├──  pseudo_state.h
│     │  │  ├──  selector.h
│     │  │  ├──  selectors.h
│     │  │  └──  state.h
│     │  ├──  syntax
│     │  │  ├──  tokenizer
│     │  │  │  └──  error.h
│     │  │  ├──  anb.h
│     │  │  ├──  base.h
│     │  │  ├──  parser.h
│     │  │  ├──  res.h
│     │  │  ├──  state.h
│     │  │  ├──  state_res.h
│     │  │  ├──  token.h
│     │  │  ├──  token_res.h
│     │  │  └──  tokenizer.h
│     │  ├──  base.h
│     │  ├──  css.h
│     │  ├──  log.h
│     │  ├──  node.h
│     │  ├──  parser.h
│     │  └──  stylesheet.h
│     ├──  dom
│     │  ├──  interfaces
│     │  │  ├──  attr.h
│     │  │  ├──  attr_const.h
│     │  │  ├──  attr_res.h
│     │  │  ├──  cdata_section.h
│     │  │  ├──  character_data.h
│     │  │  ├──  comment.h
│     │  │  ├──  document.h
│     │  │  ├──  document_fragment.h
│     │  │  ├──  document_type.h
│     │  │  ├──  element.h
│     │  │  ├──  event_target.h
│     │  │  ├──  node.h
│     │  │  ├──  processing_instruction.h
│     │  │  ├──  shadow_root.h
│     │  │  └──  text.h
│     │  ├──  base.h
│     │  ├──  collection.h
│     │  ├──  dom.h
│     │  ├──  exception.h
│     │  └──  interface.h
│     ├──  encoding
│     │  ├──  base.h
│     │  ├──  const.h
│     │  ├──  decode.h
│     │  ├──  encode.h
│     │  ├──  encoding.h
│     │  ├──  multi.h
│     │  ├──  range.h
│     │  ├──  res.h
│     │  └──  single.h
│     ├──  html
│     │  ├──  interfaces
│     │  │  ├──  anchor_element.h
│     │  │  ├──  area_element.h
│     │  │  ├──  audio_element.h
│     │  │  ├──  base_element.h
│     │  │  ├──  body_element.h
│     │  │  ├──  br_element.h
│     │  │  ├──  button_element.h
│     │  │  ├──  canvas_element.h
│     │  │  ├──  d_list_element.h
│     │  │  ├──  data_element.h
│     │  │  ├──  data_list_element.h
│     │  │  ├──  details_element.h
│     │  │  ├──  dialog_element.h
│     │  │  ├──  directory_element.h
│     │  │  ├──  div_element.h
│     │  │  ├──  document.h
│     │  │  ├──  element.h
│     │  │  ├──  embed_element.h
│     │  │  ├──  field_set_element.h
│     │  │  ├──  font_element.h
│     │  │  ├──  form_element.h
│     │  │  ├──  frame_element.h
│     │  │  ├──  frame_set_element.h
│     │  │  ├──  head_element.h
│     │  │  ├──  heading_element.h
│     │  │  ├──  hr_element.h
│     │  │  ├──  html_element.h
│     │  │  ├──  iframe_element.h
│     │  │  ├──  image_element.h
│     │  │  ├──  input_element.h
│     │  │  ├──  label_element.h
│     │  │  ├──  legend_element.h
│     │  │  ├──  li_element.h
│     │  │  ├──  link_element.h
│     │  │  ├──  map_element.h
│     │  │  ├──  marquee_element.h
│     │  │  ├──  media_element.h
│     │  │  ├──  menu_element.h
│     │  │  ├──  meta_element.h
│     │  │  ├──  meter_element.h
│     │  │  ├──  mod_element.h
│     │  │  ├──  o_list_element.h
│     │  │  ├──  object_element.h
│     │  │  ├──  opt_group_element.h
│     │  │  ├──  option_element.h
│     │  │  ├──  output_element.h
│     │  │  ├──  paragraph_element.h
│     │  │  ├──  param_element.h
│     │  │  ├──  picture_element.h
│     │  │  ├──  pre_element.h
│     │  │  ├──  progress_element.h
│     │  │  ├──  quote_element.h
│     │  │  ├──  script_element.h
│     │  │  ├──  select_element.h
│     │  │  ├──  slot_element.h
│     │  │  ├──  source_element.h
│     │  │  ├──  span_element.h
│     │  │  ├──  style_element.h
│     │  │  ├──  table_caption_element.h
│     │  │  ├──  table_cell_element.h
│     │  │  ├──  table_col_element.h
│     │  │  ├──  table_element.h
│     │  │  ├──  table_row_element.h
│     │  │  ├──  table_section_element.h
│     │  │  ├──  template_element.h
│     │  │  ├──  text_area_element.h
│     │  │  ├──  time_element.h
│     │  │  ├──  title_element.h
│     │  │  ├──  track_element.h
│     │  │  ├──  u_list_element.h
│     │  │  ├──  unknown_element.h
│     │  │  ├──  video_element.h
│     │  │  └──  window.h
│     │  ├──  tokenizer
│     │  │  ├──  error.h
│     │  │  ├──  res.h
│     │  │  ├──  state.h
│     │  │  ├──  state_comment.h
│     │  │  ├──  state_doctype.h
│     │  │  ├──  state_rawtext.h
│     │  │  ├──  state_rcdata.h
│     │  │  └──  state_script.h
│     │  ├──  tree
│     │  │  ├──  insertion_mode
│     │  │  ├──  active_formatting.h
│     │  │  ├──  error.h
│     │  │  ├──  insertion_mode.h
│     │  │  ├──  open_elements.h
│     │  │  └──  template_insertion.h
│     │  ├──  base.h
│     │  ├──  encoding.h
│     │  ├──  html.h
│     │  ├──  interface.h
│     │  ├──  interface_res.h
│     │  ├──  node.h
│     │  ├──  parser.h
│     │  ├──  serialize.h
│     │  ├──  tag.h
│     │  ├──  tag_res.h
│     │  ├──  token.h
│     │  ├──  token_attr.h
│     │  ├──  tokenizer.h
│     │  ├──  tree.h
│     │  └──  tree_res.h
│     ├──  ns
│     │  ├──  base.h
│     │  ├──  const.h
│     │  ├──  ns.h
│     │  └──  res.h
│     ├──  selectors
│     │  ├──  base.h
│     │  └──  selectors.h
│     ├──  tag
│     │  ├──  base.h
│     │  ├──  const.h
│     │  ├──  res.h
│     │  └──  tag.h
│     └──  utils
│        ├──  base.h
│        ├──  http.h
│        ├──  utils.h
│        └──  warc.h
├──  lib
│  ├──  liblexbor.so -> liblexbor.so.2
│  ├──  liblexbor.so.2 -> liblexbor.so.2.1.0
│  ├──  liblexbor.so.2.1.0
│  └──  liblexbor_static.a
└──  share
   ├──  applications
   │  └──  org.bionus.Grabber.desktop
   ├──  Grabber
   │  ├──  crashreporter
   │  │  ├──  ChineseSimplified.qm
   │  │  ├──  English.qm
   │  │  ├──  French.qm
   │  │  ├──  German.qm
   │  │  ├──  Indonesian.qm
   │  │  ├──  Italian.qm
   │  │  ├──  Japanese.qm
   │  │  ├──  Korean.qm
   │  │  ├──  Polish.qm
   │  │  ├──  Portuguese.qm
   │  │  ├──  PortugueseBrazilian.qm
   │  │  ├──  Russian.qm
   │  │  ├──  Spanish.qm
   │  │  └──  Vietnamese.qm
   │  ├──  languages
   │  │  ├──  ChineseSimplified.qm
   │  │  ├──  English.qm
   │  │  ├──  French.qm
   │  │  ├──  German.qm
   │  │  ├──  Indonesian.qm
   │  │  ├──  Italian.qm
   │  │  ├──  Japanese.qm
   │  │  ├──  Korean.qm
   │  │  ├──  languages.ini
   │  │  ├──  Polish.qm
   │  │  ├──  Portuguese.qm
   │  │  ├──  PortugueseBrazilian.qm
   │  │  ├──  Russian.qm
   │  │  ├──  Spanish.qm
   │  │  └──  Vietnamese.qm
   │  ├──  sites
   │  │  ├──  'Anime pictures'
   │  │  │  ├──  anime-pictures.net
   │  │  │  │  ├──  defaults.ini
   │  │  │  │  └──  tag-types.txt
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  ArtStation
   │  │  │  ├──  www.artstation.com
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Booru.io
   │  │  │  ├──  booru.io
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Danbooru
   │  │  │  ├──  behoimi.org
   │  │  │  │  ├──  defaults.ini
   │  │  │  │  └──  tag-types.txt
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  'Danbooru (2.0)'
   │  │  │  ├──  danbooru.donmai.us
   │  │  │  │  ├──  defaults.ini
   │  │  │  │  └──  tag-types.txt
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  DeviantArt
   │  │  │  ├──  www.deviantart.com
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  E-Hentai
   │  │  │  ├──  e-hentai.org
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  exhentai.org
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  E-Shuushuu
   │  │  │  ├──  e-shuushuu.net
   │  │  │  │  ├──  defaults.ini
   │  │  │  │  └──  tag-types.txt
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  E621
   │  │  │  ├──  e621.net
   │  │  │  │  ├──  defaults.ini
   │  │  │  │  └──  tag-types.txt
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  FurAffinity
   │  │  │  ├──  www.furaffinity.net
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  'Gelbooru (0.1)'
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  'Gelbooru (0.2)'
   │  │  │  ├──  gelbooru.com
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Gfycat
   │  │  │  └──  downloader.js
   │  │  ├──  'Hentai Cosplays'
   │  │  │  ├──  hentai-cosplays.com
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Kemono
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Mastodon
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Moebooru
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  'Newgrounds Art'
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  NHentai
   │  │  │  ├──  nhentai.net
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Nijie
   │  │  │  ├──  nijie.info
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Philomena
   │  │  │  ├──  derpibooru.org
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Pixiv
   │  │  │  ├──  www.pixiv.net
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Reddit
   │  │  │  ├──  www.reddit.com
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  'Rule34.us Gelbooru'
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  └──  sites.txt
   │  │  ├──  Shimmie
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Slushe
   │  │  │  ├──  slushe.com
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Szurubooru
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  └──  sites.txt
   │  │  ├──  Twitter
   │  │  │  ├──  api.twitter.com
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  WallHaven
   │  │  │  ├──  wallhaven.cc
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  Zerochan
   │  │  │  ├──  www.zerochan.net
   │  │  │  │  └──  defaults.ini
   │  │  │  ├──  icon.png
   │  │  │  ├──  model.js
   │  │  │  ├──  sites.txt
   │  │  │  └──  supported.txt
   │  │  ├──  exclude.txt
   │  │  ├──  exclude_xcopy.txt
   │  │  └──  helper.js
   │  ├──  themes
   │  │  ├──  Default
   │  │  │  └──  style.css
   │  │  └──  QDarkStyleSheet
   │  │     ├──  img
   │  │     │  ├──  branch_closed-on.png
   │  │     │  ├──  branch_closed.png
   │  │     │  ├──  branch_open-on.png
   │  │     │  ├──  branch_open.png
   │  │     │  ├──  checkbox_checked.png
   │  │     │  ├──  checkbox_checked_disabled.png
   │  │     │  ├──  checkbox_checked_focus.png
   │  │     │  ├──  checkbox_indeterminate.png
   │  │     │  ├──  checkbox_indeterminate_disabled.png
   │  │     │  ├──  checkbox_indeterminate_focus.png
   │  │     │  ├──  checkbox_unchecked.png
   │  │     │  ├──  checkbox_unchecked_disabled.png
   │  │     │  ├──  checkbox_unchecked_focus.png
   │  │     │  ├──  close-hover.png
   │  │     │  ├──  close-pressed.png
   │  │     │  ├──  close.png
   │  │     │  ├──  down_arrow.png
   │  │     │  ├──  down_arrow_disabled.png
   │  │     │  ├──  Hmovetoolbar.png
   │  │     │  ├──  Hsepartoolbar.png
   │  │     │  ├──  left_arrow.png
   │  │     │  ├──  left_arrow_disabled.png
   │  │     │  ├──  radio_checked.png
   │  │     │  ├──  radio_checked_disabled.png
   │  │     │  ├──  radio_checked_focus.png
   │  │     │  ├──  radio_unchecked.png
   │  │     │  ├──  radio_unchecked_disabled.png
   │  │     │  ├──  radio_unchecked_focus.png
   │  │     │  ├──  right_arrow.png
   │  │     │  ├──  right_arrow_disabled.png
   │  │     │  ├──  sizegrip.png
   │  │     │  ├──  stylesheet-branch-end.png
   │  │     │  ├──  stylesheet-branch-more.png
   │  │     │  ├──  stylesheet-vline.png
   │  │     │  ├──  transparent.png
   │  │     │  ├──  undock.png
   │  │     │  ├──  up_arrow.png
   │  │     │  ├──  up_arrow_disabled.png
   │  │     │  ├──  Vmovetoolbar.png
   │  │     │  └──  Vsepartoolbar.png
   │  │     └──  style.css
   │  ├──  webservices
   │  │  ├──  1.ico
   │  │  ├──  2.ico
   │  │  ├──  3.ico
   │  │  ├──  4.ico
   │  │  ├──  5.ico
   │  │  └──  6.ico
   │  ├──  Grabber
   │  ├──  Grabber-cli
   │  ├──  settings.ini
   │  └──  words.txt
   ├──  icons
   │  └──  hicolor
   │     ├──  48x48
   │     │  └──  apps
   │     │     └──  grabber.png
   │     ├──  96x96
   │     │  └──  apps
   │     │     └──  grabber.png
   │     └──  128x128
   │        └──  apps
   │           └──  grabber.png
   └──  metainfo
      └── 󰗀 org.bionus.Grabber.metainfo.xml
```

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 1]
```diff
@@ -35,43 +30,46 @@ stdenv.mkDerivation rec {
...
+    export HOME=$TMPDIR
+
     # the package.sh script provides some install helpers
     # using this might make it easier to maintain/less likely for the
     # install phase to fail across version bumps
-    patchShebangs ./scripts/package.sh
+    patchShebangs ../scripts/package.sh
+  '';
 
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:60]: Not only will not postPatch run now but also patches will be broken. This should all be moved to postPatch

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 2]
```diff
@@ -35,43 +30,46 @@ stdenv.mkDerivation rec {
...
     # ensure the script uses the rsync package from nixpkgs
-    substituteInPlace ../scripts/package.sh --replace "rsync" "${rsync}/bin/rsync"
+    substituteInPlace ../scripts/package.sh --replace-warn "rsync" "${rsync}/bin/rsync"
```
Comment (SuperSandro2000) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:62]:
```suggestion
    substituteInPlace ../scripts/package.sh --replace-fail "rsync" "${rsync}/bin/rsync"
```

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 3]
```diff
@@ -35,43 +30,46 @@ stdenv.mkDerivation rec {
...
     # the npm build step only runs typescript
     # run this step directly so it doesn't try and fail to download the unnecessary node_modules, etc.
-    substituteInPlace ./sites/CMakeLists.txt --replace "npm install" "npm run build"
-
-    # remove the vendored catch2
-    rm -rf tests/src/vendor/catch
-
-    # link the catch2 sources from nixpkgs
-    ln -sf ${catch2.src} tests/src/vendor/catch
+    substituteInPlace ./sites/CMakeLists.txt --replace-warn "npm install" "npm run build"
```
Comment (SuperSandro2000) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:67]:
```suggestion
    substituteInPlace ./sites/CMakeLists.txt --replace-fail "npm install" "npm run build"
```

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 4]
```diff
@@ -35,43 +30,46 @@ stdenv.mkDerivation rec {
...
+  nativeBuildInputs =
+    with qt6;
+    [
+      qtbase
+      qtdeclarative
+      qttools
+      qtnetworkauth
+      qtmultimedia
+      wrapQtAppsHook
+    ]
```
Comment (SuperSandro2000) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:42]: Those are all except the wrapQtAppsHook buildInputs

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 5]
```diff
@@ -35,43 +30,46 @@ stdenv.mkDerivation rec {
...
+    ++ [
+      makeWrapper
+      nodejs
```
Comment (SuperSandro2000) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:45]: Is this used at build time?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/im/imgbrd-grabber/package.nix (hunk 5 above)
Comment (luftmensch-luftmensch) [reply pkgs/by-name/im/imgbrd-grabber/package.nix]: Applied suggestion

**Comment by luftmensch-luftmensch** [comment]
@SuperSandro2000 applied suggestion

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 6]
```diff
@@ -1,77 +1,72 @@
...
   pname = "imgbrd-grabber";
-  version = "7.10.0";
+  version = "7.12.2";
 
   src = fetchFromGitHub {
     owner = "Bionus";
     repo = "imgbrd-grabber";
     rev = "v${version}";
-    sha256 = "sha256-AT6pN2do0LlH6xAXKcFQv+oderD88/EiG1JnCw6kOOg=";
+    sha256 = "sha256-6XfIaASfbvdPovtdDEJtsk4pEL4Dhmyq8ml4X7KZ4DE=";
```
Comment (Gerg-L) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:23]:
```suggestion
    hash = "sha256-6XfIaASfbvdPovtdDEJtsk4pEL4Dhmyq8ml4X7KZ4DE=";
```

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 7]
```diff
@@ -1,77 +1,72 @@
...
-
 stdenv.mkDerivation rec {
   pname = "imgbrd-grabber";
-  version = "7.10.0";
+  version = "7.12.2";
 
   src = fetchFromGitHub {
     owner = "Bionus";
     repo = "imgbrd-grabber";
     rev = "v${version}";
```
Comment (Gerg-L) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:22]:
```suggestion
    rev = "/refs/tags/v${finalAttrs.version}";
```

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 8]
```diff
@@ -1,77 +1,72 @@
...
+  catch2,
+  nodejs,
+  libpulseaudio,
+  openssl,
+  rsync,
+  typescript,
+  qt6,
 }:
-
 stdenv.mkDerivation rec {
```
Comment (Gerg-L) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:15]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 9]
```diff
@@ -90,11 +85,14 @@ stdenv.mkDerivation rec {
...
+    license = lib.licenses.asl20;
     homepage = "https://bionus.github.io/imgbrd-grabber/";
     mainProgram = "Grabber";
-    maintainers = [ maintainers.evanjs ];
+    maintainers = with lib.maintainers; [
+      evanjs
+      luftmensch-luftmensch
+    ];
   };
 }
```
Comment (Gerg-L) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:98]:
```suggestion
})
```

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 10]
```diff
@@ -1,77 +1,72 @@
...
     # install phase to fail across version bumps
-    patchShebangs ./scripts/package.sh
+    patchShebangs ../scripts/package.sh
+  '';
+
+  postPatch = ''
 
     # ensure the script uses the rsync package from nixpkgs
-    substituteInPlace ../scripts/package.sh --replace "rsync" "${rsync}/bin/rsync"
+    substituteInPlace ../scripts/package.sh --replace-fail "rsync" "${rsync}/bin/rsync"
```
Comment (Gerg-L) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:61]:
```suggestion
    substituteInPlace ../scripts/package.sh --replace-fail "rsync" "${lib.getExe rsync}"
```

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 11]
```diff
@@ -90,11 +85,14 @@ stdenv.mkDerivation rec {
 
   sourceRoot = "${src.name}/src";
```
Comment (Gerg-L) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:86]:
```suggestion
  sourceRoot = "${finalAttrs.src.name}/src";
```

**Review by Gerg-L — COMMENTED** [review]
Handful of little changes :D

**Comment by luftmensch-luftmensch** [comment]
@Gerg-L Applied suggestion.

**Comment by Gerg-L** [comment]
Result of `nixpkgs-review pr 312447` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>imgbrd-grabber</li>
  </ul>
</details>

**Review by Gerg-L — APPROVED** [review]

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 12]
```diff
@@ -1,77 +1,72 @@
...
 
   buildInputs = [
     openssl
     libpulseaudio
     typescript
-  ];
-
-  nativeBuildInputs = [
+    qt6.wrapQtAppsHook
     makeWrapper
```
Comment (SuperSandro2000) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:32]: makeWrapper and wrapQtAppsHook normally always go into nativeBuildInputs

File: pkgs/by-name/im/imgbrd-grabber/package.nix [hunk 13]
```diff
@@ -1,77 +1,72 @@
...
 
+  nativeBuildInputs =
+    with qt6;
+    [
+      qtbase
+      qtdeclarative
+      qttools
+      qtnetworkauth
+      qtmultimedia
+    ]
```
Comment (SuperSandro2000) [inline pkgs/by-name/im/imgbrd-grabber/package.nix:44]: Those are still not correct and should go to buildInputs.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/im/imgbrd-grabber/package.nix (hunk 13 above)
Comment (luftmensch-luftmensch) [reply pkgs/by-name/im/imgbrd-grabber/package.nix:35]: Applied suggestion. Let me know if there are other changes to make

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**